File: WebApi/Interfaces/IErpService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { RII_VW_CARI } from '../Models/RII_VW_CARI';
import { RII_VW_DEPO } from '../Models/RII_VW_DEPO';
import { RII_VW_STOK } from '../Models/RII_VW_STOK';
import { RII_VW_PROJE } from '../Models/RII_VW_PROJE';
import { RII_FN_ONHANDQUANTITY } from '../Models/RII_FN_ONHANDQUANTITY';

export interface IErpService {
  getAllCari(): Promise<ApiResponse<RII_VW_CARI[]>>;
  getAllDepo(): Promise<ApiResponse<RII_VW_DEPO[]>>;
  getAllStok(): Promise<ApiResponse<RII_VW_STOK[]>>;
  getAllProje(): Promise<ApiResponse<RII_VW_PROJE[]>>;

  getOnHandQuantityWithSerial(params: {
    depoKodu?: number;
    stokKodu?: string;
    seriNo?: string;
    projeKodu?: string;
  }): Promise<ApiResponse<RII_FN_ONHANDQUANTITY[]>>;

  healthCheck(): Promise<ApiResponse<object>>;
}

File: Interfaces/IErpService.cs
using depoWebAPI.Models;
using cms_webapi.DTOs;
using cms_webapi.Data;

namespace cms_webapi.Interfaces
{
    public interface IErpService
    {
        // VW (View) Operations - Sadece GET işlemleri
        Task<ApiResponse<IEnumerable<RII_VW_CARI>>> GetAllCariAsync();
        Task<ApiResponse<IEnumerable<RII_VW_DEPO>>> GetAllDepoAsync();
        Task<ApiResponse<IEnumerable<RII_VW_STOK>>> GetAllStokAsync();
        Task<ApiResponse<IEnumerable<RII_VW_PROJE>>> GetAllProjeAsync();

        // FN (Function) Operations - Parametreli sorgular
        Task<ApiResponse<IEnumerable<RII_FN_ONHANDQUANTITY>>> GetOnHandQuantityWithSerialAsync(
            ErpCmsDbContext context, 
            int? depoKodu = null, 
            string? stokKodu = null, 
            string? seriNo = null, 
            string? projeKodu = null);

        Task<ApiResponse<IEnumerable<RII_FN_GetProductPricing>>> GetProductPricingAsync(ErpCmsDbContext context, string stokKodu);

        // Health Check
        Task<ApiResponse<object>> HealthCheckAsync();
    }
}
